repair/repair.cc

```diff
@@ -2641,15 +2640,15 @@ future<> repair::tablet_repair_task_impl::run() {
                 auto [hints_batchlog_flushed, flush_time] = co_await rs.flush_hints(id, m.keyspace_name, tables, ignore_nodes);
                 bool small_table_optimization = false;
 
+                auto this_erm = m.erm ? std::exchange(m.erm, nullptr).release() : std::move(erm);
```

Comment:
Why I thing this needs to be refactored into sharded<>: here you can't tell by code inspection if the release() is legal or not.

Reply:
Why are m.erm and erm compatible? Is erm known to be captured after m.erm was captured?

Please add some comments to explain here.

Reply:
Outdated